### PR TITLE
Smurthi/status bug

### PR DESCRIPTION
**Pull request by swatkat1**
### Description

[Bug fix] - Update an application status to any current status. 
## Release notes

[Bug fix] - Update an application status to any current status based on latest program version and not on application's program version.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [ ] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [ ] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [ ] Created unit and/or browser tests which fail without the change (if possible)
- [ ] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

#### Database evolutions

Read the guidelines [here](https://github.com/civiform/civiform/wiki/Database#writing-database-evolutions)

- [ ] Assigned two reviewers
- [ ] Guarded against already existing resources using `IF NOT EXISTS` and `IF EXISTS`
- [ ] Downs created to undo changes in Ups

#### User visible changes

- [ ] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
  - [ ] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [ ] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

- [ ] Add new FeatureFlag env vars to `server/conf/helper/feature-flags.conf`
- [ ] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [ ] Wrote browser tests with the feature flag off and on, etc.

### Instructions for manual testing

If instructions are needed for manual testing by reviewers, include them here.

### Issue(s) this completes

Fixes #<issue_number>; Fixes #<issue_number>...

### CODE REVIEW

**Comment by swatkat1** [comment]
I think another issue might be if we end up deleting statuses, or changing their names (can we do that? I forget). Applications would end up with something in latest_status that doesn't correspond to a status in the latest program rev. By ensuring an application can only have a status that is present in the program rev it was applied for, we know this can never happen. We can maybe hack around that problem, but we're kind of creating bandaids on bandaids then. --> Today in a program, we change, ""Pending" status to "In-queue", the pending status is not completely deleted from an application. So with this fix, we are setting a status which is not initially associated with the application, but the current program version, we are not adding to any issue. Applications will still be marked "Pending". Going forwards, all new applications can be marked ""In-queue". With bulk status update, all "Pending" can be easily set to "In-queue"

File: server/app/services/applications/ProgramAdminApplicationService.java [hunk 1]
```diff
@@ -116,16 +126,12 @@ public void setStatus(
...
       }
       // Notify an Admin/TI if they applied.
       Optional<String> adminSubmitterEmail = application.getSubmitterEmail();
       if (adminSubmitterEmail.isPresent()) {
-        sendAdminSubmitterEmail(
-            programRepository.getShallowProgramDefinition(program),
-            applicant,
-            statusDef,
-            adminSubmitterEmail);
+        sendAdminSubmitterEmail(program, applicant, statusDef, adminSubmitterEmail);
```
[the remark this answers is not in the document]
Comment (swatkat1) [reply server/app/services/applications/ProgramAdminApplicationService.java:134]: This is the only reason this programDef is used -String programName = programDef.localizedName().getDefault(); I can circle back with Elise to see if is this an issue.

File: server/app/services/applications/ProgramAdminApplicationService.java [hunk 2]
```diff
@@ -136,9 +142,7 @@ public void setStatus(
         applicantEmails
             .get()
             .forEach(
-                email ->
-                    sendApplicantEmail(
-                        program.getProgramDefinition(), applicant, statusDef, Optional.of(email)));
+                email -> sendApplicantEmail(program, applicant, statusDef, Optional.of(email)));
```
[the remark this answers is not in the document]
Comment (swatkat1) [reply server/app/services/applications/ProgramAdminApplicationService.java:145]: same here. Applicants will be receiving potentially a new ProgramName from what the applied to. But my honest thought is- till the programs are fully translated, they are not shown/open to public. But I can double check.

[the remark this answers is not in the document]
Comment (avaleske) [reply server/app/services/applications/ProgramAdminApplicationService.java:145]: If the program name is changed after launch, which is less confusing:
- the applicant gets an email with the current program name?
- the applicant gets an email with the program name they applied to?

[the remark this answers is not in the document]
Comment (swatkat1) [reply server/app/services/applications/ProgramAdminApplicationService.java:145]: I checked with Elise, Matt and Miguel in our team. This is their answer - 1. The names of the programs do not change much after they are launched. They are usually generic enough to continue through reapplication process. 2. If the name is changed, they should get the latest program name not the old name.

**Pull request closed without merging**